src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,104 @@
+package duke.task;
+
+/**
+ * Represents a Task.
+ */
+public class Task {
+    protected boolean isCompleted;
+    protected String taskName;
+
+    /**
+     * Creates a Task.
+     * @param name A string representing the name of the task.
+     * @param isCompleted A boolean value representing whether the task has been completed.
+     */
+    protected Task(String name, boolean isCompleted) {
+        this.taskName = name;
+        this.isCompleted = isCompleted;
+    }
+
+    /**
+     * Creates a new Task.
+     * The completion status is false by default.
+     * @param name A string representing the name of the new task.
+     * @return The new task created.
+     */
+    public static Task newTask(String name) {
```

Comment:
createNewTask(...). Will stop commenting on this set of naming , hope that the comments will help to improve code clarity 😃